src/packet.ml

```diff
@@ -385,18 +395,27 @@ let decode_tls_data ?(with_premaster = false) buf =
      let p_len = Cstruct.BE.get_uint16 buf p_start in
      guard (Cstruct.length buf >= p_start + 2 + u_len + p_len) `Partial
      >>= fun () ->
-     maybe_string "password" buf (p_start + 2) p_len >>| fun p ->
-     let end_of_data = p_start + 2 + p_len in
-     (* for some reason there may be some slack here... *)
-     if Cstruct.length buf > end_of_data then
-       let data = Cstruct.shift buf end_of_data in
-       Log.warn (fun m ->
-           m "%u bytes slack at end of tls_data %s (p is %s)@.%a"
-             (Cstruct.length data) (Cstruct.to_string data) p Cstruct.hexdump_pp
-             data)
-     else ();
-     match (u, p) with "", "" -> None | _ -> Some (u, p))
-  >>| fun user_pass -> { pre_master; random1; random2; options; user_pass }
+     maybe_string "password" buf (p_start + 2) p_len >>= fun p ->
+     let user_pass = match (u, p) with "", "" -> None | _ -> Some (u, p) in
+     let peer_info_start = p_start + 2 + p_len in
+     (* XXX(dinosaure): if we don't have enough to have a peer-info (at least 2 bytes),
+        we just ignore it and return [None]. *)
```

Comment:
thanks for your comment. I removed the `XXX` -- which I reserve, similar as `TODO` for things that need to be done (and so a `git grep XXX` `git grep TODO` helps to figure out where to look). since this is a very valid comment, let's leave it as is :)